### DISCUSSION TITLE
Best way to check page's content before waiting for speed-limited edit

### DISCUSSION THREAD

**Discussion by hugolpz** [kanasimi/wikiapi#60] [Q&A]
My `.edit` + `.replace` is too slow (1 edit every 5 secs) :
```js
var edit = async function (title) {
        await targetWiki.edit_page(title, function(page_data) {
            return page_data.wikitext.replace(/==\n{{cc-by-sa-4.0}}/g, 
                '==\n{{Multi-license}}\n{{cc-by-sa-4.0}}\n{{cc-0}}')
        }, {bot: 1, nocreate: 1, minor: 1, summary: `Update licenses according to user's wishes, see statement in [[Special:Diff/672130985]].`});
        console.log('Done: ',JSON.stringify(list[i]));
}
```

When the script fails, it resume editing at the start of the list of 6000+ pages in my target category.

Where should I put a conditional read, so it **quicky** search if there is the target string to replace, then 
1. if there is none, go to next file as fast as possible 
2. if there is this string, replace at normal speed.

**Comment by kanasimi** [comment]
If possible, I would suggest using replace_tool, it is a well-tested tool. But I'm sorry I haven't written the documentation for that part yet, so you can only refer to the examples [here](https://github.com/kanasimi/wikibot/tree/master/replace).

In your case it would be easier to just return the Wikiapi.skip_edit. 

I also suggest you use [for_each_page](https://kanasimi.github.io/wikiapi/Wikiapi.html#for_each_page).

**Reply by hugolpz** [reply]
Thank ! I'm investigating those.

**Reply by kanasimi** [reply]
If you find that the problem has been solved, try pressing the "Mark as answer" button.

**Comment by hugolpz** [comment]
On my side, I eventually used `.skip_edit`, as suggested by Kanasimi above, and it speed up my script : 
```js
await targetWiki.edit_page(title, 
            function(page_data)
            {   // Do things:
                console.log('page_data.text: ', page_data.wikitext);
                if( page_data.wikitext.search(/==\n{{cc-by-sa-4.0}}/g) == -1){ return Wikiapi.skip_edit;}
                return page_data.wikitext.replace(/==\n{{cc-by-sa-4.0}}/g, 
                    '==\n{{Multi-license}}\n{{cc-by-sa-4.0}}\n{{cc-0}}')
            },
            {
                bot: 1, nocreate: 1, minor: 1, 
                summary: `Update licenses according to user's wishes, see statement in [[Special:Diff/672130985]].`
            }
        );
```
